config/Cargo.toml

```diff
@@ -1,24 +0,0 @@
-[package]
-name = "nexus-config"
-authors = { workspace = true }
-version = { workspace = true }
-edition = { workspace = true }
-homepage = { workspace = true }
-repository = { workspace = true }
-publish = { workspace = true }
-keywords = { workspace = true }
-categories = { workspace = true }
-
-[dependencies]
-serde = { version = "1.0", features = ["derive"] }
```

Comment:
config crate is used for various workspace configurations, and should be minimal.

An example is I want to add configs for RPC, like disconnect timeout, connections limit etc. With you PR I need to depend on `api` which bring, for example, spartan.

Work towards sdk, as I imagine it, should 
1. introduce `nexus-sdk` crate with generic trait interfaces
2. migrate workspace to using it

Reply:
I'm amenable to moving the `config` crate back out, but at that point would say I think it makes more sense to move the configuration logic to the various crates its components refer to, rather than to have this small crate dangling off to the side from the `api` also trying to combine the various functionalities spread around into one coherent interface. But if you feel strongly about keep `config` where and how it is I can move it back. 

As for the proposed flow, I agree in principle. The problem in practice is that the `prover` crate is so tied into the CLI at present that I can't make a reasonable, generic SDK that uses it in the way I'd like without doing this restructuring first.